blame/blame_test.go

```diff
@@ -0,0 +1,810 @@
+}
+
+var _ = Suite(&SuiteCommon{})
+
+var fixtureRepos = [...]struct {
+	url      string
+	packfile string
+}{
+	{"https://github.com/tyba/git-fixture.git", "../formats/packfile/fixtures/git-fixture.ofs-delta"},
+	{"https://github.com/spinnaker/spinnaker.git", "../formats/packfile/fixtures/spinnaker-spinnaker.pack"},
+}
+
+// create the repositories of the fixtures
+func (s *SuiteCommon) SetUpSuite(c *C) {
+	s.repos = make(map[string]*git.Repository, 0)
+	for _, fixRepo := range fixtureRepos {
+		s.repos[fixRepo.url] = git.NewPlainRepository()
+
+		d, err := os.Open(fixRepo.packfile)
+		defer func() {
+			c.Assert(d.Close(), IsNil)
+		}()
+		c.Assert(err, IsNil)
+
+		r := packfile.NewReader(d)
+		r.Format = packfile.OFSDeltaFormat // TODO: how to know the format of a pack file ahead of time?
+
+		_, err = r.Read(s.repos[fixRepo.url].Storage)
+		c.Assert(err, IsNil)
+	}
+}
+
+var blameTests = [...]struct {
+	// input data to revlist
+	repo   string
+	branch string // TODO: remove this, it is no longer needed for local packfiles
+	commit string
+	path   string
+	// expected output data form the revlist
+	blames []string
+}{
+	// use the blame2humantest.bash script to easily add more tests.
+	{"https://github.com/tyba/git-fixture.git", "master", "6ecf0ef2c2dffb796033e5a02219af86ec6584e5", "binary.jpg", concat(&[]string{},
+		repeat("35e85108805c84807bc66a02d91535e1e24b38b9", 285),
+	)},
+	{"https://github.com/tyba/git-fixture.git", "master", "6ecf0ef2c2dffb796033e5a02219af86ec6584e5", "CHANGELOG", concat(&[]string{},
+		repeat("b8e471f58bcbca63b07bda20e428190409c2db47", 1),
+	)},
+	{"https://github.com/tyba/git-fixture.git", "master", "6ecf0ef2c2dffb796033e5a02219af86ec6584e5", "go/example.go", concat(&[]string{},
+		repeat("918c48b83bd081e863dbe1b80f8998f058cd8294", 142),
+	)},
+	{"https://github.com/tyba/git-fixture.git", "master", "6ecf0ef2c2dffb796033e5a02219af86ec6584e5", "json/long.json", concat(&[]string{},
+		repeat("af2d6a6954d532f8ffb47615169c8fdf9d383a1a", 6492),
+	)},
+	{"https://github.com/tyba/git-fixture.git", "master", "6ecf0ef2c2dffb796033e5a02219af86ec6584e5", "json/short.json", concat(&[]string{},
+		repeat("af2d6a6954d532f8ffb47615169c8fdf9d383a1a", 22),
+	)},
+	{"https://github.com/tyba/git-fixture.git", "master", "6ecf0ef2c2dffb796033e5a02219af86ec6584e5", "LICENSE", concat(&[]string{},
+		repeat("b029517f6300c2da0f4b651b8642506cd6aaf45d", 22),
+	)},
+	{"https://github.com/tyba/git-fixture.git", "master", "6ecf0ef2c2dffb796033e5a02219af86ec6584e5", "php/crappy.php", concat(&[]string{},
+		repeat("918c48b83bd081e863dbe1b80f8998f058cd8294", 259),
+	)},
+	{"https://github.com/tyba/git-fixture.git", "master", "6ecf0ef2c2dffb796033e5a02219af86ec6584e5", "vendor/foo.go", concat(&[]string{},
+		repeat("6ecf0ef2c2dffb796033e5a02219af86ec6584e5", 7),
+	)},
+	/*
+		// Failed
+		{"https://github.com/spinnaker/spinnaker.git", "master", "f39d86f59a0781f130e8de6b2115329c1fbe9545", "InstallSpinnaker.sh", concat(&[]string{},
```

Comment:
No. I want to keep this tests commented on purpose.

These tests fail because the output of a git-blame is not unique. Right now our library and git-blame disagree on some cases. In the future we might be able to fix this discrepancies and we will need these tests cases to prove it.

Finding repos with cases like this takes a lot of time (hours), so I don't want to delete them now that I have found some.